cluster/swarm/cluster.go

```diff
@@ -574,12 +575,20 @@ func (c *Cluster) RemoveVolumes(name string) (bool, error) {
 }
 
 // Pull is exported
-func (c *Cluster) Pull(name string, authConfig *types.AuthConfig, callback func(where, status string, err error)) {
+func (c *Cluster) Pull(name string, constraints []string, authConfig *types.AuthConfig, callback func(where, status string, err error)) {
 	var wg sync.WaitGroup
 
 	c.RLock()
-	for _, e := range c.engines {
+
```

Comment:
I think this part does not have to be after `c.RLock()`. It will abuse the read lock here. @kyhavlov